src/s_sound.c

```diff
@@ -252,17 +278,40 @@ static int S_getChannel(const mobj_t *origin, sfxinfo_t *sfxinfo,
    // haleyjd 06/12/08: only if subchannel matches
    for(cnum = 0; cnum < numChannels; ++cnum)
    {
-      if(channels[cnum].sfxinfo &&
-         channels[cnum].singularity == singularity &&
-         channels[cnum].origin == origin)
-      {
-         // [FG] looping sounds don't interrupt each other
-         if (channels[cnum].sfxinfo == sfxinfo && channels[cnum].loop && loop)
+     if (!channels[cnum].sfxinfo)
+       continue;
+
+     // same sound already playing
+     if (channels[cnum].sfxinfo == sfxinfo)
+     {
+       // [FG] looping sounds don't interrupt each other
+       if (channels[cnum].origin == origin &&
+           channels[cnum].loop && loop)
+       {
+         return -1;
+       }
+
+       // Limit the number of identical sounds playing at once
+       if (++instances >= max_instances)
+       {
+         if (priority <= channels[cnum].priority)
```

Comment:
It seems `if (priority < channels[cnum].priority)` works more correct dsda_parallellimit.wad